core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -502,6 +509,9 @@ DeleteFileIndex build() {
         }
       }
 
+      if (null != globalDeletes) {
+        scanMetrics.indexedDeleteFiles().increment(globalDeletes.length);
+      }
```

Comment:
I don't think this is correct. It should be the number of delete files that were indexed for all partitions, not just the root partition. I think you should count the entries that are processed in the first loop, or add up the number of existing and added entries in the matching delete files as they are read (near `readDeleteManifest`).

Reply:
good catch as I wasn't sure if that one is correct or not. I'm counting the entries that are processed in the first loop now